File: frontend/app/types/todo.ts
import { z } from "zod";

export const todoSchema = z.object({
  id: z.number().optional(),
  user_id: z.number().optional(),
  title: z.string().min(1, { message: "タイトルは必須です" }),
  completed: z.boolean(),
  created_at: z.string().optional(),
  updated_at: z.string().optional(),
});

export type Todo = z.infer<typeof todoSchema>;

File: backend/internal/models/todo.go
// Package modelsはTodoを定義します。
package models

import (
	"time"
)

type Todo struct {
	ID        int       `json:"id,omitempty"`             // 主キー
	UserID    int       `json:"user_id"`                  // 💡 追加: ユーザーID (必須)
	Title     string    `json:"title" binding:"required"` // タスクのタイトル（必須）
	Completed bool      `json:"completed"`                // 完了状態
	CreatedAt time.Time `json:"created_at"`               // 作成日時
	UpdatedAt time.Time `json:"updated_at,omitempty"`     // 💡 追加: 更新日時
}
